### PR TITLE
feat: #182 auto detect network interface name by master ip

### PR DESCRIPTION
**Pull request by jinnzy**
### Does this pull request fix one issue?

#182

### CODE REVIEW

**Comment by allencloud** [comment]
@jinnzy please rebase your code and solve the conflict.

File: runtime/init.go [hunk 1]
```diff
@@ -209,6 +211,26 @@ func (d *Default) InitCNI() error {
 	if d.WithoutCNI {
 		return nil
 	}
+
+	// if interface is "", use master[0] network interface
+	if d.Interface != "" {
+		networkInterface, err := d.SSH.Cmd(d.Masters[0], RemoteCmdGetNetworkInterface)
+		if err != nil {
+			return fmt.Errorf("get remote network interface failed %v", err)
```
Comment (allencloud) [inline runtime/init.go:219]: fmt.Errorf("failed to get remote network interface: %v", err)

**Comment by jinnzy** [comment]
Ok

**Comment by jinnzy** [comment]
> @jinnzy please rebase your code and solve the conflict.

Submitted, please check!

File: runtime/init.go [hunk 2]
```diff
@@ -227,6 +229,26 @@ func (d *Default) InitCNI() error {
...
+		networkInterface, err := d.SSH.Cmd(d.Masters[0], RemoteCmdGetNetworkInterface)
+		if err != nil {
+			return fmt.Errorf("failed to get remote network interface: %v", err)
+		}
+		for _, v := range strings.Fields(strings.TrimSpace(string(networkInterface))) {
+			v = strings.Replace(strings.TrimSpace(v), "\n", "", -1)
+			ret, err := d.SSH.Cmd(d.Masters[0], fmt.Sprintf(RemoteCmdExistNetworkInterface, v, d.Masters[0]))
+			if err == nil {
+				if strings.Contains(string(ret), d.Masters[0]) {
+					d.Interface = v
```
Comment (fanux) [inline runtime/init.go:244]: Too many indentations, you can try to improve the logic.

File: runtime/init.go [hunk 3]
```diff
@@ -231,6 +233,27 @@ func (d *Default) InitCNI() error {
 	if d.WithoutCNI {
 		return nil
 	}
+
+	// if interface is "", use master[0] network interface
+	if d.Interface != "" {
```
Comment (fanux) [inline runtime/init.go:238]:
```suggestion
	if interface == "" {
	    using the interface witch bind to master0 IP
	} else {
	    if interface NotIn master0InterfaceList {
	       return err("failed to found xxx nic")
	    }
	}
```
I think the main logic like this is better.

**Review by fanux — APPROVED** [review]

**Pull request merged**